Weather/Services/NetworkManager.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  NetworkManager.swift
+//  Weather
+//
+//  Created by Alex on 29.07.22.
+//
+
+import Foundation
+
+class NetworkManager<T: Codable> {
+    static func fetch(for url: URL, completion: @escaping (Result<T, NetworkError>) -> Void) {
+        URLSession.shared.dataTask(with: url) { (data, response, error) in
+            guard error == nil else {
+                print(String(describing: error!))
+                completion(.failure(.error(errorMassage: error!.localizedDescription)))
+                return
+            }
+            
+            guard let httpResponse = response as? HTTPURLResponse, httpResponse.statusCode == 200 else {
+                completion(.failure(.invalidResponse))
+                return
+            }
+            
+            guard let data = data else {
+                completion(.failure(.invalidData))
+                return
+            }
+            
+            do {
+                let json = try JSONDecoder().decode(T.self, from: data)
```

Comment:
I wouldn't say it's `json`, I would rather name it `decodedObject`

Reply:
Sorry, renamed.)